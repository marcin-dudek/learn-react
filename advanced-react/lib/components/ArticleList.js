import React from 'react';
import Grid from '@material-ui/core/Grid';
import Article from './Article';

const ArticleList = props => {
  return (
    <Grid container direction="column" justify="center" alignItems="center">
      {Object.values(props.articles).map(a => (
        <Article key={a.id} article={a} getAuthor={props.getAuthor} />
      ))}
    </Grid>
  );
};

export default ArticleList;
