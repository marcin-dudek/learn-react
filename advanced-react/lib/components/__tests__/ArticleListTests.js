import React from 'react';
import ArticleList from '../ArticleList';

import renderer from 'react-test-renderer';

describe('ArticleList', () => {
  const data = {
    articles: {
      a: {id: 'a', date: '2020-04-01'},
      b: {id: 'b', date: '2020-04-01'},
    },
    getAuthor: jest.fn(_ => ({
      firstName: 'Marcin',
      lastName: 'Dudek',
      website: 'https://google.com',
    })),
  };

  it('renders correctly', () => {
    const tree = renderer.create(<ArticleList {...data} />).toJSON();

    expect(tree.children.length).toBe(2);
    expect(tree).toMatchSnapshot();
  });
});
