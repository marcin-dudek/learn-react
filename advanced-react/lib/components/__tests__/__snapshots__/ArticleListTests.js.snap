// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
>
  <div
    className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    style={
      Object {
        "marginTop": 10,
        "maxWidth": 600,
      }
    }
  >
    <div
      className="MuiCardHeader-root"
    >
      <div
        className="MuiCardHeader-avatar"
      >
        <div
          aria-label="author initials"
          className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
        >
          M
          D
        </div>
      </div>
      <div
        className="MuiCardHeader-content"
      >
        <span
          className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          2020-04-01
        </span>
      </div>
    </div>
    <div
      className="MuiCardContent-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      />
    </div>
    <div
      className="MuiCardActions-root MuiCardActions-spacing"
    >
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
        href="https://google.com"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          Marcin
           
          Dudek
        </span>
      </a>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    style={
      Object {
        "marginTop": 10,
        "maxWidth": 600,
      }
    }
  >
    <div
      className="MuiCardHeader-root"
    >
      <div
        className="MuiCardHeader-avatar"
      >
        <div
          aria-label="author initials"
          className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
        >
          M
          D
        </div>
      </div>
      <div
        className="MuiCardHeader-content"
      >
        <span
          className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          2020-04-01
        </span>
      </div>
    </div>
    <div
      className="MuiCardContent-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      />
    </div>
    <div
      className="MuiCardActions-root MuiCardActions-spacing"
    >
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
        href="https://google.com"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          Marcin
           
          Dudek
        </span>
      </a>
    </div>
  </div>
</div>
`;
